### PR TITLE
Donate PyHive to Apache Kyuubi

### PR DESCRIPTION
**Pull request by bkyryliuk**
This pr: moves https://github.com/dropbox/PyHive to the pyhive folder in apache kyuubi project at [d4ae481675ac5588ba9101596fa26f22ef0e77c4](https://github.com/dropbox/PyHive/commit/d4ae481675ac5588ba9101596fa26f22ef0e77c4) commit

and changes the license text to the standard Apache License Version 2.0.

### CODE REVIEW

**Comment by pan3793** [comment]
Many thanks, @bkyryliuk. As we discussed, let's hold on a while to wait for SGA to get approved by ASF.

**Comment by pan3793** [comment]
SGA has been received, and I have sent the IP CLEARANCE to the Incubator mailing list - https://lists.apache.org/thread/8nlbpo2ckwshsn9rvnstmzpgns2tn4fl

We can start the code merging process now

File: pyhive/README.rst [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+Project is currently unsupported
+================================
+
+
+
+
+.. image:: https://travis-ci.org/dropbox/PyHive.svg?branch=master
+    :target: https://travis-ci.org/dropbox/PyHive
+.. image:: https://img.shields.io/codecov/c/github/dropbox/PyHive.svg
+
```
Comment (pan3793) [inline pyhive/README.rst:11]: Please remove this part

```suggestion
```

File: pyhive/README.rst [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+- For Trino: `Trino installation <https://trino.io/docs/current/installation.html>`_
+- For Hive: `HiveServer2 <https://cwiki.apache.org/confluence/display/Hive/Setting+up+HiveServer2>`_ daemon
+
+Changelog
+=========
+See https://github.com/dropbox/PyHive/releases.
+
+Contributing
+============
+- Please fill out the Dropbox Contributor License Agreement at https://opensource.dropbox.com/cla/ and note this in your pull request.
```
Comment (pan3793) [inline pyhive/README.rst:159]: Please remove this line
```suggestion
```

File: pyhive/setup.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    long_description = readme.read()
+
+setup(
+    name="PyHive",
+    version=pyhive.__version__,
+    description="Python interface to Hive",
+    long_description=long_description,
+    url='https://github.com/dropbox/PyHive',
+    author="Jing Wang",
+    author_email="jing@dropbox.com",
```
Comment (pan3793) [inline pyhive/setup.py:32]:
```suggestion
    url='https://github.com/apache/kyuubi/python',
    author="Apache Kyuubi Developers",
    author_email="dev@kyuubi.apache.org",
```

**Comment by pan3793** [comment]
@bkyryliuk would you mind making the following changes? Then we can merge this patch. 

1. rename the top-level dir `pyhive` to `python`
2. add `python/**` into `.rat-exlucdes` to temporarily skip the license header check, we can fix it later.
3. as we discussed in the mailing list before, you may want to update `NOTICE` to add such info

```patch

Apache Kyuubi
Copyright 2021-2024 The Apache Software Foundation.

This product includes software developed at
The Apache Software Foundation (https://www.apache.org/).

+ (This is an example, please tune the words by yourself)
+ Portions of this software were developed at Dropbox, Inc.

...
```

**Comment by pan3793** [comment]
kindly ping @bkyryliuk, appreciate it if you could take minutes to tune this commit

**Comment by pan3793** [comment]
To unblock the ongoing development, I'm going to merge this PR as-is.

**Comment by pan3793** [comment]
Merged to master

**Comment by bkyryliuk** [comment]
@pan3793 apologies for the delay, I had no free time until recently. Is there anything else needed or are you unblocked ?

**Comment by pan3793** [comment]
@bkyryliuk thanks for the reply, you may want to update the NOTICE as I mentioned [here](https://github.com/apache/kyuubi/pull/5686#issuecomment-2026876821). Also would be good to update the PyHive README to mention the donation.

Now, we are working on setup the CI https://github.com/apache/kyuubi/issues/6281, and the next step is to finish the first PyPi release.

We are likely to create a new project(I'm going to open a discussion on the mailing list) instead of an existing PyHive. So the following steps are optional

- grant PyPi PyHive publish permission to `apachekyuubi`(`chengpan@apache.org`)
- grant PyHive repo permission to @pan3793(me) and @yaooqinn(Kent Yao, Kyuubi PMC chair)

**Pull request closed without merging**